packages/@react-stately/datepicker/src/useDateFieldState.ts

```diff
@@ -202,6 +202,8 @@ export function useDateFieldState<T extends DateValue = DateValue>(props: DateFi
     () => props.value || props.defaultValue ? {...allSegments} : {}
   );
 
+  let [clearedSegment, setClearedSegment] = useState<string>();
```

Comment:
Why is this extra state necessary? Doesn't `validSegments` already keep track when dayPeriod is cleared?

Reply:
On line [354](https://github.com/adobe/react-spectrum/pull/4648/files/18bc4c1e0e322eb03b9706a25002359402ecf909#diff-838657fffbbf57e9688a449c4899b3a29ace82a4503c9827f932f24214faa1fdR354) we remove a segment from `validSegments`. However, `setValue()` needs to know what segment was deleted for line [243](https://github.com/adobe/react-spectrum/pull/4648/files/18bc4c1e0e322eb03b9706a25002359402ecf909#diff-838657fffbbf57e9688a449c4899b3a29ace82a4503c9827f932f24214faa1fdR243) to work. When testing all the cases this came up as an edge case. Because updates are event based `setValue` has no knowledge of what triggered the update and needs this extra piece of info.

Technically this doesn't need to be state because `setValue()` is called before the state updates via `set...` and we set its value as well as calling its setter like we do for `validSegments`.

Reply:
This could probably be a ref because it doesn't need to trigger re-rendering when set.